src/Fantomas.CoreGlobalTool/Daemon.fs

```diff
@@ -30,10 +33,44 @@ type FantomasDaemon(sender: Stream, reader: Stream) as this =
 
     let exit () = disconnectEvent.Set() |> ignore
 
+    let ignoreFileStore =
+        new IgnoreFileStore<_>(FileSystem(), IgnoreList, (fun ignoreList path -> ignoreList.IsIgnored(path, false)))
+
+    let ignoreFileCancellation = new CancellationTokenSource()
+
+    let ignoreFileInvalidation =
+        let rec go () =
+            async {
+                do! Async.Sleep(TimeSpan.FromSeconds 5.0)
```

Comment:
Do we really need a time-based mechanism to clear the cache? This rubs me the wrong a bit.
Could we not parse the ignore file on demand and just keep track of the location of the ignore file.
Or we parse the ignore file and take the file's last modified timestamp into account.
We re-evaluate this timestamp and if the file is newer, we invalidate the cache.

Reply:
It's not in general possible to keep track of the location of the ignore-file, because someone might put another ignore-file nearer to the file under consideration. If we even *might* invalidate the cache on every request, then we must traverse the entire filesystem up to the cached ignore-file on every request (unless the OS can call us when a relevant file has changed).

Reply:
Which is not per se a problem, but filesystem access is really quite slow.

Reply:
That scenario of someone adding a new `.fantomasignore` file somewhere closer feels like a quite theoretical one.
It is highly unlikely to begin with that there is a `.fantomasignore` file somewhere else than the repository root.
This is an assumption but I would really like to go with it. When we first introduce the concept of the ignore file, we only checked the working directory of the cli tool.
Because that didn't work for the daemon, we started traversing parent directories until we potentially found something.
Maybe that was a bad idea and we should split up the mechanism, only traverse for the daemon (and having no cache at all) and look in the `pwd` for the regular cli tool usage.